conn.go

```diff
@@ -333,8 +344,8 @@ func (l *Conn) processMessages() {
 		for messageID, msgCtx := range l.messageContexts {
 			// If we are closing due to an error, inform anyone who
 			// is waiting about the error.
-			if l.isClosing && l.closeErr != nil {
-				msgCtx.sendResponse(&PacketResponse{Error: l.closeErr})
+			if l.isClosing() && l.closeErr.Load().(error) != nil {
```

Comment:
pretty sure it'll panic... do an uncast nil check here and cast inside the block?